OpenHPL/Waterway/AirCushionSurgeTank.mo

```diff
@@ -0,0 +1,80 @@
+  parameter Modelica.SIunits.Height h_0 = 2 "Initial water height in the surge tank" annotation (
+    Dialog(group = "Initialization"));
+  input Modelica.SIunits.Pressure p_c0 = 4*para.p_a "Initial pressure of air-cushion inside the surge tank" annotation (
+    Dialog(group = "Geometry"));
+  //// possible parameters for temperature variation. Not finished...
+  //parameter Boolean TempUse = para.TempUse "If checked - the water temperature is not constant" annotation (Dialog(group = "Initialization"));
+  //parameter Modelica.SIunits.Temperature T_i = para.T_i "Initial water temperature in the pipe" annotation (Dialog(group = "Initialization", enable = TempUse));
+  //// variables
+  Modelica.SIunits.Mass m "Water mass";
+  Modelica.SIunits.Momentum M "Water momuntum";
+  Modelica.SIunits.Force M_dot "Difference in influent and effulent momentum";
+  Modelica.SIunits.Force F "Total force acting in the surge tank";
+  Modelica.SIunits.Area A = (pi*D ^ 2) / 4 "Cross sectional area of the surge tank";
+  Modelica.SIunits.Length l = h / cos_theta "Length of water in the surge tank";
+  Real cos_theta = H / L "slope ratio";
+  Modelica.SIunits.Velocity v "Water velocity";
+  Modelica.SIunits.Force F_p "Pressure force";
+  Modelica.SIunits.Force F_f "Friction force";
+  Modelica.SIunits.Force F_g "Gravity force";
+  Modelica.SIunits.Pressure p_c "Air-cushion pressure";
+  //// initial values for differential variables
+  Modelica.SIunits.Height h(start = h_0, fixed = true) "Water height in the surge tank";
+  Modelica.SIunits.VolumeFlowRate V_dot(start = V_dot0, fixed = true) "Water flow rate";
+  //// variables for temperature. Not in use for now...
+  //Real W_f, W_e;
+  //// conector (acquisition of algebraic variable, mass flow rate m_dot, and node pressure (manifold pressure) p_n)
+  extends OpenHPL.Interfaces.ContactNode;
+initial equation
+  if SteadyState == true then
+    der(M) = 0;
+    der(m) = 0;
+    //der(T_n) = 0;
+  else
+    h = h_0;
+    V_dot = V_dot0;
+    //T_n = T_i;
+  end if;
+equation
+  //// mass and momentum balance
+  der(m) = m_dot;
+  der(M) = M_dot+F;
+
+  m = para.rho * A * l;
+  m_dot = para.rho*V_dot;
+
+  M = m * v;
+  v = V_dot / A;
+  M_dot = m_dot*v;
+  F = F_p-F_f-F_g;
+  F_p = (p_n - p_c)*A;
+  p_c = p_c0*((L-h_0/cos_theta)/(L-l))^para.gamma_air;
+  F_f = Functions.DarcyFriction.Friction(v, D, l, para.rho, para.mu, eps);
+  F_g = m * para.g * cos_theta;
+ annotation (
+    Documentation(info= "<html>
+<p>The simple model of the air-cushion surge tank, which is described by the momentum and mass balance differential equations. The mass balance depends on inlet and outlet mass flow rates. The momentum balance depends on inlet momentum and pressure dorp through the surge pipe together with gravity and friction forces. The main defined variable are <i>V_dot</i> and <i>h  </i>(the flow rate and water level in the surge tank). The air in the surge tank is taken at inital pressure of <i>p_c0.</i></p>
```

Comment:
When revering to code variables you should use `<code>X_y</code>` instead of `<i>`. Actually `<i>` should not be used at all. If you want to emphasise use `<em>` instead.